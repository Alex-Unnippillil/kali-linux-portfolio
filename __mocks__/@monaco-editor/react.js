import React from 'react';
export default function MonacoMock({ value = '', onChange }) {
  return React.createElement('textarea', {
    value,
    onChange: e => onChange && onChange(e.target.value),
  });
}
