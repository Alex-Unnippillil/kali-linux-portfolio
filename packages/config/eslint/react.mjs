import globals from 'globals';
import { baseConfig, compat } from './base.mjs';

const reactConfig = [
  ...baseConfig,
  {
    languageOptions: {
      globals: globals.browser,
    },
    rules: {
      'no-top-level-window/no-top-level-window-or-document': 'off',
      'no-undef': 'off',
      'no-unused-vars': 'off',
      'react/react-in-jsx-scope': 'off',
    },
  },
  ...compat.config({
    extends: [
      'eslint:recommended',
      'plugin:@typescript-eslint/recommended',
      'plugin:react/recommended',
      'plugin:react-hooks/recommended',
    ],
    settings: {
      react: {
        version: 'detect',
      },
    },
  }),
  {
    rules: {
      '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
      '@typescript-eslint/no-unused-expressions': ['error', { allowShortCircuit: true }],
      'react/react-in-jsx-scope': 'off',
    },
  },
];

export default reactConfig;
