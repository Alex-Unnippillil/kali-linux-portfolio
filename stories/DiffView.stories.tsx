import React from 'react';

import DiffView from '../components/common/DiffView';

const meta = {
  title: 'Common/DiffView',
  component: DiffView,
  parameters: {
    layout: 'centered',
  },
};

export default meta;

const Template = (args: React.ComponentProps<typeof DiffView>) => <DiffView {...args} />;

export const TextDiff = Template.bind({});
TextDiff.args = {
  leftLabel: 'Baseline',
  rightLabel: 'Update',
  mode: 'text',
  left: 'Kali Linux Portfolio',
  right: 'Kali Linux Portfolio v2',
};

export const JsonDiff = Template.bind({});
JsonDiff.args = {
  leftLabel: 'Current',
  rightLabel: 'Incoming',
  mode: 'json',
  left: {
    name: 'Terminal',
    permissions: ['read', 'write'],
    theme: 'dark',
    pinned: true,
  },
  right: {
    name: 'Terminal',
    permissions: ['read', 'write', 'share'],
    theme: 'dark',
    pinned: false,
    telemetry: true,
  },
};
