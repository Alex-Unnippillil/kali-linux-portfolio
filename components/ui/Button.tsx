"use client";

import { forwardRef, type ButtonHTMLAttributes } from "react";
import clsx from "clsx";

const variantStyles = {
  primary:
    "bg-ubt-blue text-white hover:bg-ubt-blue/90 focus-visible:outline-ubt-blue disabled:bg-ubt-blue/70",
  secondary:
    "bg-ub-cool-grey text-white hover:bg-ub-cool-grey/90 focus-visible:outline-ub-cool-grey disabled:bg-ub-cool-grey/70",
  ghost: "bg-transparent text-white hover:bg-white/10 focus-visible:outline-white disabled:text-white/70",
  destructive:
    "bg-ub-red text-white hover:bg-ub-red/90 focus-visible:outline-ub-red disabled:bg-ub-red/70",
} as const;

const sizeStyles = {
  sm: "px-3 py-1.5 text-sm",
  md: "px-4 py-2 text-sm",
  lg: "px-5 py-3 text-base",
} as const;

export type ButtonVariant = keyof typeof variantStyles;
export type ButtonSize = keyof typeof sizeStyles;

export type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  variant?: ButtonVariant;
  size?: ButtonSize;
  loading?: boolean;
};

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      className,
      variant = "primary",
      size = "md",
      loading = false,
      disabled,
      children,
      type,
      ...props
    },
    ref,
  ) => {
    const isDisabled = Boolean(disabled || loading);

    return (
      <button
        ref={ref}
        type={type ?? "button"}
        className={clsx(
          "relative inline-flex items-center justify-center rounded-md border border-transparent font-medium transition-colors",
          "focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2",
          "disabled:cursor-not-allowed disabled:opacity-60 data-[loading=true]:cursor-wait",
          "min-h-[32px] min-w-[32px] select-none whitespace-nowrap",
          variantStyles[variant],
          sizeStyles[size],
          loading && "opacity-80",
          className,
        )}
        disabled={isDisabled}
        aria-busy={loading || undefined}
        data-loading={loading || undefined}
        {...props}
      >
        {loading && (
          <span
            aria-hidden
            className="mr-2 inline-flex h-4 w-4 items-center justify-center"
          >
            <span className="h-4 w-4 animate-spin rounded-full border-2 border-current border-t-transparent" />
          </span>
        )}
        {children}
      </button>
    );
  },
);

Button.displayName = "Button";

export default Button;
