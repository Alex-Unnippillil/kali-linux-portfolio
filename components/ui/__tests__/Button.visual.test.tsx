import { act, render } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../Button.stories';

describe('Button stories visual regression', () => {
  const composedStories = composeStories(stories);

  Object.entries(composedStories).forEach(([name, Story]) => {
    it(`${name} matches the snapshot`, () => {
      const { container, getByRole } = render(<Story />);

      if (name === 'FocusVisible') {
        act(() => {
          getByRole('button').focus();
        });
      }

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
