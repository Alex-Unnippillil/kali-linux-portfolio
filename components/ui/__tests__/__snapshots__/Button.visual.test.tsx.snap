// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button stories visual regression Destructive matches the snapshot 1`] = `
<button
  class="relative inline-flex items-center justify-center rounded-md border border-transparent font-medium transition-colors focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed disabled:opacity-60 data-[loading=true]:cursor-wait min-h-[32px] min-w-[32px] select-none whitespace-nowrap bg-ub-red text-white hover:bg-ub-red/90 focus-visible:outline-ub-red disabled:bg-ub-red/70 px-4 py-2 text-sm"
  type="button"
>
  Delete
</button>
`;

exports[`Button stories visual regression Disabled matches the snapshot 1`] = `
<button
  class="relative inline-flex items-center justify-center rounded-md border border-transparent font-medium transition-colors focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed disabled:opacity-60 data-[loading=true]:cursor-wait min-h-[32px] min-w-[32px] select-none whitespace-nowrap bg-ubt-blue text-white hover:bg-ubt-blue/90 focus-visible:outline-ubt-blue disabled:bg-ubt-blue/70 px-4 py-2 text-sm"
  disabled=""
  type="button"
>
  Disabled
</button>
`;

exports[`Button stories visual regression FocusVisible matches the snapshot 1`] = `
<button
  class="relative inline-flex items-center justify-center rounded-md border border-transparent font-medium transition-colors focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed disabled:opacity-60 data-[loading=true]:cursor-wait min-h-[32px] min-w-[32px] select-none whitespace-nowrap bg-ubt-blue text-white hover:bg-ubt-blue/90 focus-visible:outline-ubt-blue disabled:bg-ubt-blue/70 px-4 py-2 text-sm"
  type="button"
>
  Focus Visible
</button>
`;

exports[`Button stories visual regression Ghost matches the snapshot 1`] = `
<button
  class="relative inline-flex items-center justify-center rounded-md border border-transparent font-medium transition-colors focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed disabled:opacity-60 data-[loading=true]:cursor-wait min-h-[32px] min-w-[32px] select-none whitespace-nowrap bg-transparent text-white hover:bg-white/10 focus-visible:outline-white disabled:text-white/70 px-4 py-2 text-sm"
  type="button"
>
  Ghost Button
</button>
`;

exports[`Button stories visual regression Large matches the snapshot 1`] = `
<button
  class="relative inline-flex items-center justify-center rounded-md border border-transparent font-medium transition-colors focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed disabled:opacity-60 data-[loading=true]:cursor-wait min-h-[32px] min-w-[32px] select-none whitespace-nowrap bg-ubt-blue text-white hover:bg-ubt-blue/90 focus-visible:outline-ubt-blue disabled:bg-ubt-blue/70 px-5 py-3 text-base"
  type="button"
>
  Large Button
</button>
`;

exports[`Button stories visual regression Loading matches the snapshot 1`] = `
<button
  aria-busy="true"
  class="relative inline-flex items-center justify-center rounded-md border border-transparent font-medium transition-colors focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed disabled:opacity-60 data-[loading=true]:cursor-wait min-h-[32px] min-w-[32px] select-none whitespace-nowrap bg-ubt-blue text-white hover:bg-ubt-blue/90 focus-visible:outline-ubt-blue disabled:bg-ubt-blue/70 px-4 py-2 text-sm opacity-80"
  data-loading="true"
  disabled=""
  type="button"
>
  <span
    aria-hidden="true"
    class="mr-2 inline-flex h-4 w-4 items-center justify-center"
  >
    <span
      class="h-4 w-4 animate-spin rounded-full border-2 border-current border-t-transparent"
    />
  </span>
  Loading
</button>
`;

exports[`Button stories visual regression Primary matches the snapshot 1`] = `
<button
  class="relative inline-flex items-center justify-center rounded-md border border-transparent font-medium transition-colors focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed disabled:opacity-60 data-[loading=true]:cursor-wait min-h-[32px] min-w-[32px] select-none whitespace-nowrap bg-ubt-blue text-white hover:bg-ubt-blue/90 focus-visible:outline-ubt-blue disabled:bg-ubt-blue/70 px-4 py-2 text-sm"
  type="button"
>
  Click me
</button>
`;

exports[`Button stories visual regression Secondary matches the snapshot 1`] = `
<button
  class="relative inline-flex items-center justify-center rounded-md border border-transparent font-medium transition-colors focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed disabled:opacity-60 data-[loading=true]:cursor-wait min-h-[32px] min-w-[32px] select-none whitespace-nowrap bg-ub-cool-grey text-white hover:bg-ub-cool-grey/90 focus-visible:outline-ub-cool-grey disabled:bg-ub-cool-grey/70 px-4 py-2 text-sm"
  type="button"
>
  Secondary Button
</button>
`;

exports[`Button stories visual regression Small matches the snapshot 1`] = `
<button
  class="relative inline-flex items-center justify-center rounded-md border border-transparent font-medium transition-colors focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 disabled:cursor-not-allowed disabled:opacity-60 data-[loading=true]:cursor-wait min-h-[32px] min-w-[32px] select-none whitespace-nowrap bg-ubt-blue text-white hover:bg-ubt-blue/90 focus-visible:outline-ubt-blue disabled:bg-ubt-blue/70 px-3 py-1.5 text-sm"
  type="button"
>
  Small Button
</button>
`;
